File: apps/android/app/src/main/java/com/example/shop/feature/cart/data/model/CartItem.kt
package com.example.shop.feature.cart.data.model

import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonObject
import kotlinx.serialization.json.buildJsonObject

@Serializable
data class CartAddRequest(val skuId: String, val quantity: Int)

@Serializable
data class CartUpdateRequest(val skuId: String, val quantity: Int)

@Serializable
data class CartRemoveRequest(val skuIds: List<String>)

@Serializable
data class CartSelectRequest(val skuIds: List<String>, val selected: Boolean)

// ── API response models ──

@Serializable
data class CartItemResponse(
    val skuId: String,
    val quantity: Int,
    val selected: Boolean,
    val snapshot: CartItemSnapshot,
    val currentPrice: String,
    val currentStock: Int,
    val priceChanged: Boolean = false,
    val unavailable: Boolean = false,
    val stockInsufficient: Boolean = false,
)

@Serializable
data class CartItemSnapshot(
    val productId: String,
    val productTitle: String,
    val skuAttrs: JsonObject = buildJsonObject {},
    val price: String = "0",
    val imageUrl: String? = null,
)

// ── Domain model used by UI ──

data class CartItem(
    val skuId: String,
    val quantity: Int,
    val selected: Boolean,
    val productId: String,
    val productTitle: String,
    val price: Double,
    val stock: Int,
    val attributes: JsonObject,
    val imageUrl: String? = null,
)

File: apps/ios/Shop/Feature/Cart/Data/Model/CartItem.swift
import Foundation

struct CartItem: Codable, Equatable, Identifiable, Sendable {
    let skuId: String
    var quantity: Int
    var selected: Bool
    let addedAt: String?
    let snapshot: CartSnapshot
    let currentPrice: String
    let currentStock: Int
    let priceChanged: Bool
    let unavailable: Bool
    let stockInsufficient: Bool

    var id: String { skuId }

    var price: Double {
        Double(currentPrice) ?? 0
    }

    var subtotal: Double {
        price * Double(quantity)
    }
}

struct CartSnapshot: Codable, Equatable, Sendable {
    let productId: String
    let productTitle: String
    let skuAttrs: [String: String]?
    let price: String
    let imageUrl: String?
}
